### PR TITLE
Pre shield

### PR DESCRIPTION
**Pull request by jonathanKingston**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1105,22 +1093,50 @@ ContainerWindow.prototype = {
...
+
+    // Only hack the normal plus button as the alltabs is done elsewhere
+    this.attachMenuEvent("plus-button", this._plusButton);
+
     attachTo(this._style, this._window);
   },
 
+  attachMenuEvent(source, button) {
+    const popup = button.querySelector(".new-tab-popup");
+    popup.addEventListener("popupshown", (event) => {
```
Comment (groovecoder) [inline index.js]: `  1113:43  error  'event' is defined but never used  no-unused-vars`

Comment (jonathanKingston) [inline index.js]: Arse I fixed that.... just never pushed

**Comment by groovecoder** [comment]
Looks good and the 2 new pings show up in `about:telemetry` when I use the old long-press + UI.

Do we want to merge this now? If we merge it into master, it will go into Test Pilot with the next Test Pilot update. Or should we make the shield `.xpi` off of this branch?

File: index.js [hunk 2]
```diff
@@ -1105,22 +1093,50 @@ ContainerWindow.prototype = {
...
+
+    // Only hack the normal plus button as the alltabs is done elsewhere
+    this.attachMenuEvent("plus-button", this._plusButton);
+
     attachTo(this._style, this._window);
   },
 
+  attachMenuEvent(source, button) {
+    const popup = button.querySelector(".new-tab-popup");
+    popup.addEventListener("popupshown", () => {
```
Comment (groovecoder) [inline index.js]: Crap. On Fx 54, this triggers `TypeError: popup is null`

Comment (groovecoder) [inline index.js]: Also affects 55. Looks like 56+ is fine?

Comment (groovecoder) [inline index.js]: Filed https://github.com/mozilla/testpilot-containers/issues/658

File: index.js [hunk 3]
```diff
@@ -1437,11 +1341,6 @@ ContainerWindow.prototype = {
     }
   },
 
-  _shutdownPlusButtonMenu() {
```
Comment (groovecoder) [inline index.js]: This caused `TypeError: this._shutdownPlusButtonMenu is not a function` during a 2.3.0 -> 2.4.0 upgrade on Fx 56.

**Pull request merged**